### DISCUSSION TITLE
First Start: Offer a Tour

### DISCUSSION THREAD

**Discussion by kojid0** [element-hq/element-meta#91] [Ideas]
### Your use case

It is important that users can use the app as quickly as possible after downloading it. This means that the number of questions at the first start should be reduced to a minimum.
Still, it is important that users understand how Element works and what (layout) settings they want to have.

A tour can
- show feature highlights of element while explaining them (what is 'decental', what is 'home/identity server', what is a 'room', what is 'space',...)
- show how to find contacts to chat with (otherwise users will delete the app right away when none of their phone book contacts is shown on Element)
- ask whether you want to be found publicly via mobile phone number / email (#2228 / vector-im/element-android#3791)
- ask about the preferred layout (#650 / vector-im/element-android#4010)
- ask to give rating on the app store (#3876)
- ...

**Comment by HarHarLinks** [comment]
> * show feature highlights of element while explaining them (what is 'decental', what is 'home/identity server', what is a 'room', what is 'space',...)

disagree: users will be familiar with what a chat and a group/room is. only the stuff essential to using the app should be explained proactively in your face like this. while explanation for more stuff should be easy to find, it should appear only when relevant (e.g. "learn more about spaces" when first opening the space panel, a `?` button in it, or similar). opening the settings -> help menu should explain everything in detail (incl decentralized, identity server, ...), perhaps give the full tutorial as you explain, but only when users go looking for it.

> * show how to find contacts to chat with (otherwise users will delete the app right away when none of their phone book contacts is shown on Element)

This is assuming this is an empty account, if it's not they will already know the concepts. Clicking the "speech bubble plus" button in the DMs tab will show a search field and buttons "invite friends", "contacts book", and "qr code". The really rough edge I see here is that it looks much different when clicking the "#+" button in the rooms tab. This also raises the question of how to handle this in "simple" design of vector-im/element-android#4010. also see the next point:

> * ask whether you want to be found publicly via mobile phone number / email ([phone nr - based registration and login #2228](https://github.com/vector-im/element-android/issues/2228) / [Start new chat via Contacts Book: Ask if user wants to share phone number  #3791](https://github.com/vector-im/element-android/issues/3791))

not quite sure about this one, it might seem too much.

i wonder how important 3PIDs are at all? are there metrics? Afaik none of my contacts have email nor phone number added to their account.

> * ask about the preferred layout ([Display conversations all in one list #650](https://github.com/vector-im/element-android/issues/650) / [Navigation | New Home tab to priorize all unread #4010](https://github.com/vector-im/element-android/pull/4010))

agree

> * ask to give rating on the app store ([Ask active users to leave us a review on the app store #3876](https://github.com/vector-im/element-android/issues/3876))

disagree

---

> important that users can use the app as quickly as possible

While a short into to the app is great, there should be a "i don't care, skip all" button that populates with the more simple choice where settings are concerned. Also some intro slides may be skipped depending on whether the account is new/empty.

**Comment by Vnzq4Qje3zzU** [comment]
Adding that https://github.com/vector-im/element-android/issues/979 could get integrated to it too. 

https://github.com/vector-im/element-android/issues/979#issuecomment-584594932:
> Matrix often is used for "real"/professional information exchange (like Slack) - that's another use-case than WhatsApp-like smalltalk to me. Current style fits both, bubble-style fits more the last example. thinking

I was thinking a lot about it and maybe a good separation for the design choice is between the "workspace mode" (à la discord/slack) which gives a lot of information in the screen like how it's done now and an "instant messaging mode" (à la WhatsApp/Telegram) which focus on simplicity and makes it more friendly to everyone.

About the option of the user being able to skip everything, I think it's a good idea overall, but most of the users won't mind clicking 2 or 3 extra buttons, specially in mobile clients and with people wanting to use for personal conversations, where most of them will only set up it again when they buy a new smartphone.

**Comment by daniellekirkwood** [comment]
Converting this issue to a discussion in our Element-Meta repo as it's not platform specific and the ideas included in this issue so far would benefit all platforms. 

We're currently working on our onboarding flow and will incorporate your ideas where able and appropriate - thanks so much for the feedback! Please keep adding more thoughts.

**Comment by kittykat** [comment]
Hi all! If anyone is interested in following development work around adding a tour/wizzard, the Delight team have taken on [this issue](https://github.com/vector-im/element-web/issues/1631) and you can subscribe there for updates.

Thank you all for your input so far!
